Documentation: Add referenced links to existing documentation (#20898)
fleetdm/fleet: docs/Deploy/Reference-Architectures.md

## The Fleet binary

> The Fleet application contains two single static binaries which provide web based administration, REST API, and [CLI interface](https://fleetdm.com/guides/fleetctl) to Fleet.

Reviewer: ```suggestion
The Fleet application contains two single static binaries which provide web based administration, a REST API, and a [CLI interface](https://fleetdm.com/guides/fleetctl).
```

> The Fleet application contains two single static binaries which provide web based administration, REST API, and [CLI interface](https://fleetdm.com/docs/using-fleet/fleetctl-cli) to Fleet.

Reviewer 2: @JoGSal Sorry about this, the page that this links to has been moved to the guides section of the website. Can you update this URL?

```suggestion
The Fleet application contains two single static binaries which provide web based administration, REST API, and [CLI interface](https://fleetdm.com/guides/fleetctl) to Fleet.
```

Author: Absolutely, thank you for pointing it out!

## The Fleet binary

The Fleet application contains two single static binaries which provide web based administration, REST API, and [CLI interface](https://fleetdm.com/docs/using-fleet/fleetctl-cli) to Fleet.

The `fleet` binary contains:
- The [Fleet TLS web server](https://fleetdm.com/docs/configuration/fleet-server-configuration) (no external webserver is required but it supports a proxy if desired)
- The Fleet web interface
- The Fleet application management [REST API](https://fleetdm.com/docs/using-fleet/rest-api)

> - The [Fleet osquery API endpoints](https://fleetdm.com/docs/rest-api/rest-api) which enables FleetDM's functionality powered by osquery

Reviewer 2: Can you remove this change? The bullet point above it is already has a link to the REST API documentation.
```suggestion
- The Fleet osquery API endpoints
```

Author: Well spotted @eashaw thank you. I would've rather kept the text after the link but I hope the update as requested works
